Add GitHub CI build action

ngc7293 opened the pull request:
Fixes #1

ngc7293 committed:
Ajoute une action de CI on: push

* `shared` est maintenant inclus comme submodule
* mise-a-jour du Makefile
* ajout du fichier de configuration de CI pour GitHub Actions

ngc7293 committed:
Small teak to build.yml

ngc7293 committed:
Updates Makefile to treat warning as errors

This is still aerospace

ngc7293 committed:
Fixes uninitialized in sd_diskio.c, update shared

le-Bark commented on Src/sd_diskio.c:
```diff
@@ -262,7 +262,8 @@ DRESULT SD_read(BYTE lun, BYTE *buff, DWORD sector, UINT count)
     /* Slow path, fetch each sector a part and memcpy to destination buffer */
     int i;
-    uint8_t ret;
+    uint8_t ret = MSD_ERROR;
```
C'est du code généré par cubeMX non?

ngc7293 replied:
Oui, mais le warning de GCC est valide selon moi. Je préfère faire la correction manuellement plutôt que de désactivé le warning.

ngc7293 committed:
Change initial value of ret

Reading the code more closely I realized the only context where `ret`
would be read uninitialized is if count == 0, in which case we should
probably mark it as a success.

ngc7293 committed:
Update shared to point to master again

(PR was merged in anirniq-shared)
